pxr/imaging/hdx/unitTestDelegate.cpp

```diff
@@ -58,7 +58,49 @@ TF_DEFINE_PRIVATE_TOKENS(
     _tokens,
 
     (renderBufferDescriptor)
-);    
+);
+
+
+// XXX: These mostly-empty subclasses are here to prevent a link-time
+// optimization with HdRenderIndex::InsertTask<T>(), hdx/taskController.cpp and
+// HD_TRACE_FUNCTION() for address sanitized builds of USD. Without these
+// classes the linker will incorrectly "optimize" calls to the expanded
+// HD_TRACE_FUNCTION() in GetRenderIndex()->InsertTask<HdxPickTask>() for
+// taskController.cpp.
+class Hdx_UnitTestPickTask : public HdxPickTask {
```

Comment:
I welcome feedback here as I'm not sure what the best solution is. Effectively, `GetRenderIndex()->InsertTask<HdxPickTask>()` is defined in `unitTestDelegate.cpp` and `taskController.cpp`. The linker will, understandably, optimize out the symbol in `taskController.o` since `unitTestDelegate.o` defines it first. This causes a linker error since it's `GetRenderIndex()->InsertTask<HdxPickTask>()` has been instrumented with ASan inside of `taskController.cpp`